lib/krikri/mapper.rb

```diff
@@ -62,6 +62,37 @@ def map(name, records)
     ##
     # An application-wide registry of defined mappings
     Registry = Class.new(Krikri::Registry)
-    
+
+    ##
+    # A SoftwareAgent to run mapping processes.
+    #
+    # @see: Krikri::SoftwareAgent, Krikri::Activity
+    class Agent
+      include SoftwareAgent
+
+      attr_reader :name, :generator_uri
+
+      def initialize(opts = {})
+        @name = opts.fetch(:name).to_sym
+        @generator_uri = RDF::URI(opts.fetch(:generator_uri))
+      end
+
+      def run(activity_uri = nil)
+        Krikri::Mapper.map(name, records).each do |rec|
+          rec.set_subject!(SecureRandom.hex)
```

Comment:
I want to have a conversation with the team about how to handle identifiers and which component should have the responsibility of assigning them.

I have considered doing this in the mapping (via the existing `#uri` DSL method), but don't have a plan for handling mappings that fail to set a URI.  A URI is necessary to save a `DPLA::MAP::Aggregation` as an LDP resource, so it's insufficient to allow objects w/out URI's to remain blank nodes.